### PR TITLE
Add HomeKit service

### PR DESCRIPTION
**Pull request by bertrandda**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

I added service to integrate HomKit protocol to Gladys.
Gladys instance become a bridge HomeKit compatible and allow access devices through Apple Home app & Siri.

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@ static
 gladys-backups
 qemu-*
 coverage
+persist
```
Comment (Pierre-Gilles) [inline .gitignore:15]: What is persist?

File: server/package.json [hunk 2]
```diff
@@ -38,7 +38,8 @@
   "license": "Apache-2.0",
   "nodemonConfig": {
     "ignore": [
-      "test/*"
+      "test/*",
+      "persist/*"
```
Comment (Pierre-Gilles) [inline server/package.json:42]: persist?

File: server/services/homekit/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   * @returns {Promise} Empty promise once service is started.
+   * @example
+   * gladys.services.homekit.start();
+   */
+  async function start() {
+    logger.info('Starting HomeKit service');
+    const dockerBased = await gladys.system.isDocker();
+    if (dockerBased) {
+      const { basePathOnContainer } = await gladys.system.getGladysBasePath();
+      hap.HAPStorage.setCustomStoragePath(`${basePathOnContainer}/homekit`);
```
Comment (Pierre-Gilles) [inline server/services/homekit/index.js:22]: Could you cover this with tests? You can mock gladys.system.isDocker to return true

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for this PR 🙏 I did a review and have just a few comments, nothing really serious.

After your changes I'll merge this PR 🙂

Thanks again for this !

File: .gitignore (hunk 1 above)
Comment (bertrandda) [reply .gitignore:15]: `persist/` is a folder created by the lib `hap-node` to store bridge and accessories configuration, in prod (with docker) it stored in `/var/lib/gladysassistant/homekit` (define here https://github.com/GladysAssistant/Gladys/pull/1591/files#diff-e49a6d70a5acaa68174761271b7d850f6971bfaab7d0d150912f411acc585ebbR22). In dev it create persist/ folder in server root folder

File: server/package.json (hunk 2 above)
Comment (bertrandda) [reply server/package.json:42]: Following previous explanation about persist folder, it avoids nodemon server reload loop on bridge configuration (when HomeKit integration is starting it write files in this folder)

File: .gitignore (hunk 1 above)
Comment (Pierre-Gilles) [reply .gitignore:15]: Good for me! Thanks :)

**Review by Pierre-Gilles — APPROVED** [review]
Good for me! Thanks for this PR! 🙏

**Pull request merged**